backend/src/v4/models/alluxioClient.js

```diff
@@ -0,0 +1,160 @@
+/**
+ *  Copyright (C) 2020 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+const axios = require("axios");
+
+const axiosJSONConfig = {
+	headers: {
+		"Content-Type": "application/json;charset=UTF-8",
+		"Access-Control-Allow-Origin": "*"
+	}
+};
+
+class AlluxioError extends Error {
+	constructor(msg, innerError) {
+		super(msg);
+		this.innerError = innerError;
+	}
+}
+
+class AlluxioClient {
+	constructor(hostname) {
+		this.hostname = hostname;
+	}
+
+	getURL(basePath) {
+		return `http://${this.hostname}/api/v1/${basePath}`;
+	}
+
+	getPathsURL(action, path) {
+		return `${this.getURL("paths")}/${path}/${action}`;
+	}
+
+	getStreamsURL(action, id) {
+		return `${this.getURL("streams")}/${id}/${action}`;
+	}
+
+	async postToPathRoute(path, opts, action) {
+		return (await axios.post(this.getPathsURL(action, path), opts, axiosJSONConfig)).data;
+	}
+
+	async postToStreamRoute(id, opts, action) {
+		return (await axios.post(this.getStreamsURL(action, id), opts, axiosJSONConfig)).data;
+	}
+
+	/**
+	 * Creates a directory
+	 *
+	 * @param {string} path - The absolute directory path to be created.
+	 * @param {string} opts - The absolute directory path to be created.
+	 */
+	async createDirectory(path, opts) {
+		return await this.postToPathRoute(path, opts, "create-directory");
```

Comment:
either these don't need to be async, or it doesn't need to have return.

i.e. 
```suggestion
async createDirectory(path, opts) {
    await this.postToPathRoute(path, opts, "create-directory");
}
```

or

```suggestion
createDirectory(path, opts) {
   return this.postToPathRoute(path, opts, "create-directory");
}
````